config/routes/admin.rb

```diff
@@ -69,7 +69,9 @@
     post '/product_import/save_data', to: 'product_import#save_data', as: 'product_import_save_async'
     post '/product_import/reset_absent', to: 'product_import#reset_absent_products', as: 'product_import_reset_async'
 
-    resources :dfc_product_imports, only: [:index]
+    resources :dfc_product_imports, only: [:index] do
+      post :import, on: :collection
+    end
```

Comment:
I see the temptation to call the action import but I would prefer to go with Rails defaults, convention over configuration. In this domain, I would say that we `create` a Product Import. Following that logic, the index action should probably be a `new` action. :thinking: But the form on the product import screen could also be seen the the `new` form.

What do you think?

Reply:
I didn't think a lot about it, but thought that Rails' resource/collection model didn't fit well in this case, so simply opted to copy the existing action name on the `ProductImportController`. 
I suppose another way to look at it is an action on the products resource itself, ie: `/admin/products/import` or `/admin/products/batch_create`. 

Ultimately I don't think it's worth thinking about anymore ;)
But it would be good to have a third opinion.

Reply:
To further clarify my thinking, I didn't think there was much "cost" at all using a non-standard action name.
And I think to "create a Product Import" sounds too abstract, which makes it harder to see what it's for.

Reply:
When possible I prefer to stick with the defaults Rails action, but I think in this case David's solution is acceptable. After all, we can't really refer to Updating or Deleting a "Product Import", we don't have a tangible model for "Product Import".